test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -0,0 +1,859 @@
+        $m4 = 'System.Collections.Generic.IList`1[System.Management.Automation.PSTypeName] InferTypeOf(System.Management.Automation.Language.Ast, System.Management.Automation.PowerShell, System.Management.Automation.TypeInferenceRuntimePermissions)'
+
+        $inferTypeOf1 = $inferType.Where{$m1 -eq $_}[0]
+        $inferTypeOf2 = $inferType.Where{$m2 -eq $_}[0]
+        $inferTypeOf3 = $inferType.Where{$m3 -eq $_}[0]
+        $inferTypeOf4 = $inferType.Where{$m4 -eq $_}[0]
+
+        class AstTypeInference {
+            static [IList[PSTypeName]] InferTypeOf([Language.Ast] $ast) {
+                return  $script:inferTypeOf1.Invoke($null, $ast)
+            }
+
+            static [IList[PSTypeName]] InferTypeOf([Language.Ast] $ast, [System.Management.Automation.TypeInferenceRuntimePermissions] $runtimePermissions) {
+                return $script:inferTypeOf2.Invoke($null, @($ast, $runtimePermissions))
+            }
+
+            static [IList[PSTypeName]] InferTypeOf([Language.Ast] $ast, [System.Management.Automation.PowerShell] $powershell) {
+                return $script:inferTypeOf3.Invoke($null, @($ast, $powershell))
+            }
+
+            static [IList[PSTypeName]] InferTypeOf([Language.Ast] $ast, [PowerShell] $powerShell, [System.Management.Automation.TypeInferenceRuntimePermissions] $runtimePermissions) {
+                return $script:inferTypeOf4.Invoke($null, @($ast, $powerShell, $runtimePermissions))
+            }
+        }
+
+    }
+
+    It "Inferes type from integer" {
+        $res = [AstTypeInference]::InferTypeOf( { 1 }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.Int32'
+    }
+
+
+    It "Inferes type from string literal" {
+        $res = [AstTypeInference]::InferTypeOf( { "Text" }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.String'
+    }
+
+    It "Inferes type from type expression" {
+        $res = [AstTypeInference]::InferTypeOf( { [int] }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.Type'
+    }
+
+    It "Inferes type from hashtable" {
+        $res = [AstTypeInference]::InferTypeOf( { @{} }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.Collections.Hashtable'
+    }
+
+    It "Inferes type from array expression" {
+        $res = [AstTypeInference]::InferTypeOf( { @() }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.object[]'
+    }
+
+    It "Inferes type from Array literal" {
+        $res = [AstTypeInference]::InferTypeOf( { , 1 }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.object[]'
+    }
+
+    It "Inferes type from array IndexExpresssion" {
+        $res = [AstTypeInference]::InferTypeOf( { (1, 2, 3)[0] }.Ast)
+        $res.Count | Should Be 1
+        $res.Name | Should be 'System.object'
```

Comment:
Do you expect me to fix this in PR?
I don't have a todo for this. I just tried to capture the current general behavior of arrays in a test.